engine/gamesys/src/gamesys/scripts/script_sound.cpp

```diff
@@ -536,14 +536,16 @@ namespace dmGameSystem
 
         uint32_t play_id = dmSound::GetAndIncreasePlayCounter();
 
-        int functionref = 0;
+        // We use UINTPTR_MAX value to determine if it's a function call without a callback or a `play_sound` message call from Lua.
+        // Take a look at comp_sound -> CompSoundOnMessage for a better understanding of how it's used.
+        uintptr_t functionref = UINTPTR_MAX;
```

Comment:
I'm not super thrilled using a single variable for two different things, but I think we can use it like this for now.

A more proper way would be to alter the actual message, with a flag set to indicate the how the sound was invoked.

Reply:
Yeah. We already use this approach with play_id, so I decided it's fine to do so here as well.